src/mono/mono/mini/debugger-agent.c

```diff
@@ -7163,14 +6719,15 @@ vm_commands (int command, int id, guint8 *p, guint8 *end, Buffer *buf)
 		tls->pending_invoke->endp = tls->pending_invoke->p + (end - p);
 		tls->pending_invoke->suspend_count = suspend_count;
 		tls->pending_invoke->nmethods = nmethods;
-
-		if (flags & INVOKE_FLAG_SINGLE_THREADED) {
-			resume_thread (THREAD_TO_INTERNAL (thread));
-		}
-		else {
-			count = suspend_count;
-			for (i = 0; i < count; ++i)
-				resume_vm ();
+		if (!CHECK_PROTOCOL_VERSION(3, 0)) {
+			if (flags & INVOKE_FLAG_SINGLE_THREADED) {
```

Comment:
this is because icordbg sends a separate resume command after invoke?

Reply:
Yes.